test_data/asm/book/instructions.asm

```diff
@@ -40,11 +40,17 @@ machine Main {
     instr add X, Y -> Z = submachine.add; // - trivial usage, equivalent to:
                                           //   instr add X, Y -> Z = submachine.add X, Y -> Z;
 // ANCHOR_END: trivial
-    instr add_to_A X, Y = submachine.add X, Y -> A; // - output to a regular register
+    instr add_to_A X, Y = submachine.add X, Y -> A';// - output to a regular register
```

Comment:
Whether to use `A` or `A'` is actually up to the user, both work as "expected".
If you use `A` as output then you probably can't use it when calling the instruction as input!
Consider the example using poseidon from `coprocessors.rs`:
```
instr poseidon_gl = poseidon_gl.poseidon_permutation P0, P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11 -> P0, P1, P2, P3;
```
This is allowed! But won't work because input can't be equal to output in the _same line_. What we want here is output into the next line.

Reply:
The output can also be a complex expression which itself includes registers... maybe the user does want to reference the current value from some register there? What should we allow or not? I think for now we leave the user responsible for writing the lookup he wants.
Also, this might be useful for implementing instruction delegation, where one instruction outputs to the same line, so that the following instruction uses that output as input (imaginary syntax ahead):
```
instr foo_into_C X,Y
    = arith.mul X,3 -> A,
    = arith.mul Y,5 -> B,
{
    C' = A + B,
}
```